### PR TITLE
WELD-1991 Fix invocation of overloaded method on decorator

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

**Comment by weld-tester** [comment]
Triggering build using a merge of 952528b829eb31c746d4fb2d5587c1d3ac94726d on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 918 is now running using a merge of 952528b829eb31c746d4fb2d5587c1d3ac94726d on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/918

**Comment by weld-tester** [comment]
Build 918 outcome was SUCCESS using a merge of 952528b829eb31c746d4fb2d5587c1d3ac94726d on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/918
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/918

File: impl/src/main/java/org/jboss/weld/bean/DecoratedMethods.java [hunk 1]
```diff
@@ -62,17 +67,37 @@
...
-        return NULL_MARKER;
+        if (matching.isEmpty()) {
+            // No match
+            return NULL_MARKER;
+        } else if (matching.size() == 1) {
+            // If there is only one matching method no further action is needed
+            return matching.get(0);
+        }
+        // Choose the most specific method
+        // This does meet all requirements of JLS but it should work in most cases
```
Comment (jharting) [inline impl/src/main/java/org/jboss/weld/bean/DecoratedMethods.java]: I do not fully understand the comment. Perhaps it is wrong? If this code meets all the JLS requirements then it should work in all (not most) cases, right?

**Comment by weld-tester** [comment]
Build 918 outcome was SUCCESS using a merge of 952528b829eb31c746d4fb2d5587c1d3ac94726d on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/918
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/918

**Comment by weld-tester** [comment]
Build 918 outcome was SUCCESS using a merge of 952528b829eb31c746d4fb2d5587c1d3ac94726d on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/918
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/918

**Comment by weld-tester** [comment]
Build 918 outcome was SUCCESS using a merge of 952528b829eb31c746d4fb2d5587c1d3ac94726d on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/918
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/918

File: impl/src/main/java/org/jboss/weld/bean/DecoratedMethods.java (hunk 1 above)
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bean/DecoratedMethods.java]: `not` is missing. I'll fix the comment.

**Comment by weld-tester** [comment]
Triggering build using a merge of e3d6be62041b923bb0d1373597948f9910bb272a on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 940 is now running using a merge of e3d6be62041b923bb0d1373597948f9910bb272a on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/940

**Comment by jharting** [comment]
Merged, thanks!

**Pull request closed without merging**